### PR TITLE
Add Default Export Path to ExportCommand

### PR DESCRIPTION
**Pull request by LivingOnEdge**
Issue #129: Enables the user to export memes to a default path by using p/d for the path prefix in the ExportCommand

### CODE REVIEW

File: src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java [hunk 1]
```diff
@@ -28,6 +29,9 @@
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_FILEPATH + "C:/Users/username/Downloads/meme_folder/ ";
     public static final String MESSAGE_SUCCESS = "Memes exported successfully!";
+    public static final String MESSAGE_INVALID_DEFAULT_EXPORT_PATH =
+            "Invalid Default Export Path set. "
+            + "Configure it in preferences.json and restart the application";
```
[the remark this answers is not in the document]
Comment (LivingOnEdge) [reply src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java]: Yes, it will be generated at the directory where the jar file is

File: src/main/java/seedu/weme/logic/parser/commandparser/exportcommandparser/ExportCommandParser.java [hunk 2]
```diff
@@ -30,9 +31,17 @@ public ExportCommand parse(String args) throws ParseException {
             throw new ParseException(MESSAGE_MISSING_PATH);
         }
 
-        DirectoryPath path = ParserUtil.parseDirectoryPath(argMultimap.getValue(PREFIX_FILEPATH).get());
-
-        return new ExportCommand(path);
+        String directoryPath = argMultimap.getValue(PREFIX_FILEPATH).get();
+        ExportCommand command;
+        switch (directoryPath) {
+        case DEFAULT_DIRECTORY:
```
Comment (jonchan51) [inline src/main/java/seedu/weme/logic/parser/commandparser/exportcommandparser/ExportCommandParser.java]: Would it make more sense to have the empty argument `export` command input be the default directory export instead?
Like `export p/asdasd` will export to folder `asdasd`
`export` will export to default

File: src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java [hunk 3]
```diff
@@ -54,6 +65,17 @@ public CommandResult execute(Model model) throws CommandException {
         return new CommandResult(MESSAGE_SUCCESS);
     }
 
+    public DirectoryPath getExportPath(Model model) throws CommandException {
```
Comment (jonchan51) [inline src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java]: Might be more intuitive to just return a Path object here instead of a DirectoryPath object

File: src/main/java/seedu/weme/logic/parser/commandparser/exportcommandparser/ExportCommandParser.java (hunk 2 above)
Comment (moziliar) [reply src/main/java/seedu/weme/logic/parser/commandparser/exportcommandparser/ExportCommandParser.java]: Makes sense is makes sense. But what if the user wanted to export to a certain directory and then accidentally press enter after typing export. He has to spend sometime finding what’s the default path then go delete the folder. Btw I tried the default path. It shows it’s to my root path...

File: src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (moziliar) [reply src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java]: Hmm. Let me try again. Before I commented, the path I read is my root directory.

File: src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java (hunk 3 above)
Comment (LivingOnEdge) [reply src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java]: True but DirectoryPath does validation of the directory path so it is a security measure.

Comment (jonchan51) [reply src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java]: if so then wouldnt it be better to just call the DirectoryPath's isValid method rather than converting the whole thing into a DirectoryPath object?

Comment (LivingOnEdge) [reply src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java]: It is not a utility class so we should minimize coupling. I will remove the staticness of the isValid method.

Comment (jonchan51) [reply src/main/java/seedu/weme/logic/commands/exportcommand/ExportCommand.java]: All our data classes like Description have a static isValid method though, changing this is actually deviating from the original design. In this case since you are creating a DirectoryPath object, the coupling is the same

**Comment by moziliar** [comment]
LGTM
